UnityProject/Assets/Scripts/HealthV2/Living/BodyParts/RootBodyPartContainer.cs

```diff
@@ -352,7 +352,7 @@ public virtual void SubBodyPartAdded(BodyPart implant)
 			//This is so you can still hit for example the Second Head of a double-headed thing, can be changed if we find a better solution for aiming at Specific body parts
 			if (damageSplit || attackType == AttackType.Bomb || attackType == AttackType.Fire || attackType == AttackType.Rad)
 			{
-				foreach (var ContainsLimb in ContainsLimbs)
+				foreach (var ContainsLimb in ContainsLimbs.ToArray())
```

Comment:
what type were these enumrables before the casting to Array/List?

Reply:
I have no idea but all of these caused bombs to break when exploding near players so I had to do this to fix this issue.
No they explode normally again without breaking.

Reply:
Line 68
public List<BodyPart> ContainsLimbs = new List<BodyPart>();

Reply:
foreach should work fine with either arrays or lists, any enumerable actually. I'm trying to think how this change fixed the problem or if it in fact did anything.

Reply:
it did do something. It works now and doesn't throw errors so it fixed it. Test it yourself.
https://stackoverflow.com/questions/604831/collection-was-modified-enumeration-operation-may-not-execute

Reply:
the issue is that, the bomb would cause a limb to get removed but it's applying the damage still so it breaks the list it's iterating over

Reply:
Yes, and this is the fix for it.

Reply:
so the concern we have now is generating a considerable amount of garbage when you create a new list every time someone takes damage. We're discussing solutions at the moment.

Reply:
the solution I would choose would have a list on the health master base that you had the limb/organ that you want to be cut off , then it loops through that once it's finished taking damage

Reply:
Yes I know I'm late on this, but you're wrong about how this works Bod.
We're trying to damage the limb, not cut it off.

Creating another list to track a separate list is terrible as well, I've been looking for a solution for the past few hours and all what stackoverflow and the internet has told me is that the only way to avoid this error is to just use `ToArray()`
There is no way to escape this I believe.

Reply:
creating a separate list once vs creating a list every time someone takes damage. That's the thing.

Reply:
Except that this list will be useless 99% of the time and will be sitting there in memory for no reason + it has a very specific case that no other function or object will use so this is basically the same as the problem of creating a public variable that only one function uses it and has no place anywhere else.

Don't even get me started on having to maintain it and make sure that it's synced with the original list so the game doesn't throw a fit when it detects that a bodypart is no longer inside that list.

Reply:
so instead of making the list one time you think it is better to create new lists every time someone takes damage and when it is no longer used, make the GC dispose it at the cost of performance?

Reply:
No. That's why I'm looking for a solution right now that's actually good for performance.
Creating a list that's going to be hell to maintain and synced with the original list and is taking up memory all game is not good, nor is using this method right now.

Reply:
FOUND THE FIX
HAHA

Reply:
Do I understand correctly that the problem was that when an explosion damage the limbs in `foreach()`, some of them can be removed from the list(limbs fly off), so due to the severing of the limbs, the iterations are interrupted and the damage stops being dealt?

I know I'm late with this question, but I would still like to know the answer to it, as I have some suspicions about the solution to this problem in recent commits.

Reply:
You're correct and yeah everyone is a lil sus about my solution because of index changes.
I'm trying other solutions now.